Core/GDCore/Extensions/Metadata/BehaviorMetadata.h

```diff
@@ -257,7 +256,21 @@ class GD_CORE_API BehaviorMetadata {
    * \note An empty string means the base object, so any object.
    */
   const gd::String& GetObjectType() const { return objectType; }
-#endif
+
+  /**
+   * Check if the behavior is private - it can't be used outside of its
+   * extension.
+   */
+  bool IsPrivate() const { return isPrivate; }
+
+  /**
+   * Set that the behavior is private - it can't be used outside of its
+   * extension.
+   */
+  BehaviorMetadata &SetPrivate() {

```

Comment:
Should this function take a boolean as parameter? So that one can revert the isPrivate status

Reply:
In C++, this can even be a default parameter (default parameters don't work when exposed in JS, but nice to have in C++).

Reply:
Actually, these class are used declaratively, it won't be mutated much. It's public by default and it can be declared private but it will never need to change after this. The next generation will use new instances.